lib/src/extended_resize_image_provider.dart

```diff
@@ -0,0 +1,231 @@
+///
+/// The decoded image may still be displayed at sizes other than the
+/// cached size provided here.
+///
+/// [scaling] If not null, the original image's width and height will
+/// divide with it to calculate targetWidth.
+/// [maxBytes] If not null, the image will resize to a size that nolarger
+/// than [maxBytes]. Default size is 500KB.
+class ExtendedResizeImage extends ImageProvider<_SizeAwareCacheKey>
+    with ExtendedImageProvider<_SizeAwareCacheKey> {
+  const ExtendedResizeImage(
+    this.imageProvider, {
+    this.scaling,
+    this.maxBytes = 500 << 10,
+    this.width,
+    this.height,
+    this.allowUpscaling = false,
+  })  : assert(scaling != null ||
+            maxBytes != null ||
+            width != null ||
+            height != null),
+        assert(allowUpscaling != null);
+
+  /// The [ImageProvider] that this class wraps.
+  final ImageProvider imageProvider;
+
+  /// The maxBytes the image should decode to and cache.
+  final int? maxBytes;
+
+  /// The scaling the image should decode to and cache.
+  final double? scaling;
+
+  /// The width the image should decode to and cache.
+  final int? width;
+
+  /// The height the image should decode to and cache.
+  final int? height;
+
+  /// Whether the [width] and [height] parameters should be clamped to the
+  /// intrinsic width and height of the image.
+  ///
+  /// In general, it is better for memory usage to avoid scaling the image
+  /// beyond its intrinsic dimensions when decoding it. If there is a need to
+  /// scale an image larger, it is better to apply a scale to the canvas, or
+  /// to use an appropriate [Image.fit].
+  final bool allowUpscaling;
+
+  /// Composes the `provider` in a [ResizeImage] only when `cacheWidth` and
+  /// `cacheHeight` are not both null.
+  ///
+  /// When `cacheWidth` and `cacheHeight` are both null, this will return the
+  /// `provider` directly.
+  ///
+  /// Extended with `scaling` and `maxBytes`.
+  static ImageProvider<Object> resizeIfNeeded(
+    int? cacheWidth,
+    int? cacheHeight,
+    ImageProvider<Object> provider, {
+    double? scaling,
+    int? maxBytes,
+  }) {
+    if (cacheWidth != null ||
+        cacheHeight != null ||
+        scaling != null ||
+        maxBytes != null) {
+      return ExtendedResizeImage(
+        provider,
+        width: cacheWidth,
+        height: cacheHeight,
+        maxBytes: maxBytes,
+        scaling: scaling,
+      );
+    }
+    return provider;
+  }
+
+  @override
+  ImageStreamCompleter load(_SizeAwareCacheKey key, DecoderCallback decode) {
+    final DecoderCallback decodeResize = (Uint8List bytes,
+        {int? cacheWidth, int? cacheHeight, bool? allowUpscaling}) {
+      assert(
+          cacheWidth == null && cacheHeight == null && allowUpscaling == null,
+          'ResizeImage cannot be composed with another ImageProvider that applies '
+          'cacheWidth, cacheHeight, or allowUpscaling.');
```

Comment:
Add the trailing comma then format again.

```suggestion
      assert(
        cacheWidth == null && cacheHeight == null && allowUpscaling == null,
        'ResizeImage cannot be composed with another ImageProvider that applies '
        'cacheWidth, cacheHeight, or allowUpscaling.',
      );
```